FourthHomework28.09.21/TelephoneDirectory/TelephoneDirectory.c

```diff
@@ -0,0 +1,219 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <locale.h>
+#include <string.h>
+
+typedef struct
+{
+    char phone[20];
+    char name[30];
+}TelephoneDerectory;
+
+bool checkThatAnArraysAreIdentical(const char firstArray[],
+    const char secondArray[], const int lengthOfArrays)
+{
+    for (int i = 0; i < lengthOfArrays; ++i)
+    {
+        if (firstArray[i] != secondArray[i])
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+int writeTelephoneDerectoyIntoFile(const char fileName[], 
+    const TelephoneDerectory userDataBase[], const int lengthOfBase)
+{
+    FILE* file = fopen(fileName, "w");
+    for (int i = 0; i < lengthOfBase; ++i)
+    {
+        fwrite(userDataBase[i].phone, sizeof(char), 20, file);
+        fwrite(" ", sizeof(char), 1, file);
+        fwrite(userDataBase[i].name, sizeof(char), 30, file);
+        fwrite("\n", sizeof(char), 1, file);
+    }
+    fclose(file);
+    return 0;
+}
+
+void stringCopy(char mainArray[], char secondArray[])
+{
+    for (int i = 0; secondArray[i] != '\0'; ++i)
+    {
+        mainArray[i] = secondArray[i];
+    }
+}
+
+void printCharArray(const char theArray[], const int lengthOfTheArray)
+{
+    for (int i = 0; i < lengthOfTheArray; ++i)
+    {
+        printf("%c", theArray[i]);
+    }
+}
+
+int findName(char desiredPhone[], const TelephoneDerectory userDataBase[], 
```

Comment:
```suggestion
int findName(const char desiredPhone[], const TelephoneDerectory userDataBase[], 
```
Чтобы можно было спокойно строковые литералы передавать. Ну и вообще, это ведь функция-запрос, она не должна менять свои аргументы.